contracts/src/cosmos/precompile/governance.sol

```diff
@@ -119,4 +112,29 @@ interface IGovernanceModule {
         string noCount;
         string noWithVetoCount;
     }
+
+    /**
+     * @dev Emitted by the governance module when `submitProposal` is called.
+     */
+    event SubmitProposal(uint64 indexed proposalId, string indexed proposalMessages);
+
+    /**
+     * @dev Emitted by the governance module when `submitProposal` is called.
+     */
+    event SubmitProposal(uint64 indexed votingPeriodStart);
+
+    /**
+     * @dev Emitted by the governance module when `submitProposal` is called.
+     */
+    event ProposalDeposit(string indexed amount, uint64 indexed proposalId);
```

Comment:
amount should probably not be indexed and is a uint256 right?
```suggestion
    event ProposalDeposit(uint64 indexed proposalId, uint256 amount);
```

Reply:
its a uint64, no need to use up to 256 since cosmos modules can only be max uint64

Reply:
should be indexed though, and amount should be a uint256 since it has to be an amount of the gov token on the chain so shouldnt have to do a string.

Reply:
why would the amount of proposal deposit be indexed on? adds unnecessary overhead. imo variables to index by are usually IDs, start/end times, from/to addresses, we don't index on amount for any of the bank/staking precompiles

Reply:
we ended up removing the deposit amount since it would override the bank send event which uses the same key, `AttributeKeyAmount`